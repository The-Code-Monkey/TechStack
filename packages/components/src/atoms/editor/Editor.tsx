import EditorContainer from "@techstack/lexical";

import "@techstack/lexical/css";

export const Editor = ({ name }) => {
    return (
        <div>
            <EditorContainer name={name} value={''} onChange={() => {}} />
            <EditorContainer name={name} value={'<p>test</p>'}  onChange={() => {}} />
            <EditorContainer name={name} value={''}  onChange={() => {}} />
        </div>
    )
}

export default Editor;