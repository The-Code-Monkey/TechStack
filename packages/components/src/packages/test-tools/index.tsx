import { mount, shallow, ShallowWrapper, ReactWrapper } from 'enzyme';
import React, { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import theme from '../theme';
import { ConfigContext } from '../utils';

const config = {
  outputDir: '/src/packages/',
  iconDir: 'feather',
};

const removeProperties = () => {
  const keys = ['theme', 'styledTheme'];
  return {
    test: (val: any) =>
      val &&
      typeof val === 'object' &&
      'props' in val &&
      Object.keys(val.props).some((prop) => keys.some((key) => key === prop)),
    print: (val: any, serialize: any) => {
      keys.forEach((key) => {
        delete val.props[key];
      });
      return serialize(val);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors?.common,
    modes: theme.colors?.modes,
    ...theme.colors?.modes.light,
  },
};

export const shallowWithTheme: (children: ReactNode) => ShallowWrapper = (
  children: ReactNode
) =>
  shallow(<ThemeProvider theme={formattedTheme}>{children}</ThemeProvider>)
    .dive()
    .shallow();

export const mountWithTheme: (children: ReactNode) => ReactWrapper = (
  children: ReactNode
) => {
  expect.addSnapshotSerializer(removeProperties());

  return mount(
    <ConfigContext.Provider value={config}>
      <ThemeProvider theme={formattedTheme}>
        <>{children}</>
      </ThemeProvider>
    </ConfigContext.Provider>
  );
};
