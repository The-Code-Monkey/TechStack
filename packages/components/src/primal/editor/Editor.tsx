import ReactQuill from '@techstack/react-quill';

import '@techstack/react-quill/dist/quill.snow.css';

export const Editor = () => {
  return (
    <div>
      <ReactQuill />
    </div>
  );
};

export default Editor;
