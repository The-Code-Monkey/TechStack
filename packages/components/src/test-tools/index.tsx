import isPropValid from '@emotion/is-prop-valid';
import { render, RenderResult } from '@testing-library/react';
import { ReactNode } from 'react';
import { StyleSheetManager, ThemeProvider } from 'styled-components';
import { expect } from 'bun:test';

import theme from '../theme';
import { ConfigContext } from '../utils';

const config = {
  outputDir: '/src/packages/',
  iconDir: 'feather',
};

const removeProperties = () => {
  const keys = ['theme', 'styledTheme'];
  return {
    test: (val: { props: unknown }) =>
      val &&
      typeof val === 'object' &&
      'props' in val &&
      Object.keys(val.props).some(prop => keys.some(key => key === prop)),
    serialize: (val, config, indentation, depth, refs, printer) => {
      keys.forEach(key => {
        delete val.props[key];
      });
      return printer(val, config, indentation, depth, refs);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors?.common,
    modes: theme.colors?.modes,
    ...theme.colors?.modes.light,
  },
};

export const mountWithTheme = (children: ReactNode): RenderResult => {
  // expect.addSnapshotSerializer(removeProperties());

  return render(
    <StyleSheetManager shouldForwardProp={isPropValid}>
      <ConfigContext.Provider value={config}>
        <ThemeProvider theme={formattedTheme}>
          <>{children}</>
        </ThemeProvider>
      </ConfigContext.Provider>
    </StyleSheetManager>
  );
};
