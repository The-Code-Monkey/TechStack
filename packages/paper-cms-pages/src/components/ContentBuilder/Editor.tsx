import { Box } from '@techstack/components';
import ReactQuill from 'react-quill';
import 'react-quill/dist/quill.core.css';
import 'react-quill/dist/quill.snow.css';

import { EditorWrapper, StyledAccordion } from './styled';

interface Props {
  value: string;
  onChange: (e: any) => void;
  name?: string;
}

const Editor = ({ value, onChange, name }: Props) => {
  const handleOnChange = (value?: string) => {
    onChange({
      target: {
        name,
        value,
      },
    });
  };

  return (
    <Box d='flex' flexDir='column' w='full'>
      <StyledAccordion
        title={
          value
            ?.split('</h1>')[0]
            .replace(/(<[a-zA-Z0-9]>|<\/[a-zA-Z0-9]>)/g, '')
            .replace('&amp;', '&') ?? 'No Content'
        }
        maxHeight={7000}
      >
        <EditorWrapper>
          <ReactQuill theme={'snow'} value={value} onChange={handleOnChange} />
        </EditorWrapper>
      </StyledAccordion>
    </Box>
  );
};

export default Editor;
