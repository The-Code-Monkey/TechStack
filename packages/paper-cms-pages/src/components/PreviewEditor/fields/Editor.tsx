import { useCallback, useState } from 'react';
import ReactQuill from 'react-quill';
import 'react-quill/dist/quill.core.css';

import { DefaultRecordType } from '../../../utils/types';
import { EditorWrapper } from '../styled';

interface Props {
  item: DefaultRecordType;
  styles: Record<string, string>;
  onChange: (index: string, value: string) => void;
  onHoverChange: (hoveredElement: string, isTop: boolean) => void;
  hoveredElement: [string, boolean];
  onReorder: (thisId: string, thatId: string, top: boolean) => void;
}
const Editor = ({
  item,
  styles,
  onChange,
  onHoverChange,
  hoveredElement: [hoveredId, isTop],
  onReorder,
}: Props) => {
  const [isDragging, setIsDragging] = useState(false);

  const handleOnChange = (index: string) => (value: string) => {
    onChange(index, value);
  };

  const findCorrectTarget = useCallback(
    (parentTarget: any): any => {
      return parentTarget.className.includes(item.id)
        ? parentTarget
        : findCorrectTarget(parentTarget.parentElement);
    },
    [item.id]
  );

  const handleOnDragOver = useCallback(
    (e: any) => {
      if (!isDragging) {
        e.preventDefault();
        let target = e.target;
        if (!target.className.includes(item.id)) {
          target = findCorrectTarget(target.parentElement);
        }

        const box = target.getBoundingClientRect();
        const offsetTop = e.clientY - box.top - box.height / 2;

        const id = target.className
          .split(' ')
          .find((str: string) => str.startsWith('#'))
          .split('_')[1];

        onHoverChange(id, offsetTop < 0);
      }
    },
    [findCorrectTarget, isDragging, item.id, onHoverChange]
  );

  const handleOnDragEnd = () => {
    setIsDragging(false);
    onReorder(item.id, hoveredId, isTop);
  };

  const className = `innerSection_${item.id}`;

  const isHovered = hoveredId === item.id;

  return (
    <EditorWrapper
      {...(styles ?? {})}
      draggable
      onDragOver={handleOnDragOver}
      onDragStart={() => setIsDragging(true)}
      onDragEnd={handleOnDragEnd}
      className={`#${className}`}
      isTop={isTop}
      isHovered={isHovered}
    >
      <ReactQuill
        theme={false as any}
        value={item.value}
        onChange={handleOnChange(item.id)}
      />
    </EditorWrapper>
  );
};

export default Editor;
