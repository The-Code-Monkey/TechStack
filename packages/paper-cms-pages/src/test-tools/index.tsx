import { ConfigContext } from '@techstack/components';
import { render } from '@testing-library/react';
import { ReactNode } from 'react';

import theme from '../theme';
import { ThemeProvider } from '../workarounds/styled-components';

const config = {
  outputDir: '/src/packages/',
  iconDir: 'feather',
};

const removeProperties = () => {
  const keys = ['theme', 'styledTheme'];
  return {
    test: (val: { props: unknown }) =>
      val &&
      typeof val === 'object' &&
      'props' in val &&
      Object.keys(val.props).some(prop => keys.some(key => key === prop)),
    serialize: (val, config, indentation, depth, refs, printer) => {
      keys.forEach(key => {
        delete val.props[key];
      });
      return printer(val, config, indentation, depth, refs);
    },
  };
};

const formattedTheme = {
  ...theme,
  colors: {
    common: theme.colors?.common,
    modes: theme.colors?.modes,
    ...theme.colors?.modes.light,
  },
};

export const mountWithTheme = (children: ReactNode) => {
  expect.addSnapshotSerializer(removeProperties());

  return render(
    <ConfigContext.Provider value={config}>
      <ThemeProvider theme={formattedTheme}>
        <>{children}</>
      </ThemeProvider>
    </ConfigContext.Provider>
  );
};
