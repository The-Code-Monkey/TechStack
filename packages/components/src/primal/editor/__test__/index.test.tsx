import { mountWithTheme } from '../../../test-tools';
import { Editor } from '../Editor';

describe('<Box />', () => {
  it('renders children correctly', () => {
    const { asFragment } = mountWithTheme(<Editor />);
    expect(asFragment()).toMatchSnapshot();
  });
});
