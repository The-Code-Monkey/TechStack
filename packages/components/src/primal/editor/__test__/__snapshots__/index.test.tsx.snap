// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> renders children correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="quill "
    >
      <div
        class="ql-toolbar ql-snow"
      >
        <span
          class="ql-formats"
        >
          <span
            class="ql-header ql-picker"
          >
            <span
              aria-controls="ql-picker-options-0"
              aria-expanded="false"
              class="ql-picker-label"
              role="button"
              tabindex="0"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <polygon
                  class="ql-stroke"
                  points="7 11 9 13 11 11 7 11"
                />
                 
                <polygon
                  class="ql-stroke"
                  points="7 7 9 5 11 7 7 7"
                />
                 
              </svg>
            </span>
            <span
              aria-hidden="true"
              class="ql-picker-options"
              id="ql-picker-options-0"
              tabindex="-1"
            >
              <span
                class="ql-picker-item"
                data-value="1"
                role="button"
                tabindex="0"
              />
              <span
                class="ql-picker-item"
                data-value="2"
                role="button"
                tabindex="0"
              />
              <span
                class="ql-picker-item"
                data-value="3"
                role="button"
                tabindex="0"
              />
              <span
                class="ql-picker-item ql-selected"
                role="button"
                tabindex="0"
              />
            </span>
          </span>
          <select
            class="ql-header"
            style="display: none;"
          >
            <option
              value="1"
            />
            <option
              value="2"
            />
            <option
              value="3"
            />
            <option
              selected="selected"
            />
          </select>
        </span>
        <span
          class="ql-formats"
        >
          <button
            class="ql-bold"
            type="button"
          >
            <svg
              viewBox="0 0 18 18"
            >
               
              <path
                class="ql-stroke"
                d="M5,4H9.5A2.5,2.5,0,0,1,12,6.5v0A2.5,2.5,0,0,1,9.5,9H5A0,0,0,0,1,5,9V4A0,0,0,0,1,5,4Z"
              />
               
              <path
                class="ql-stroke"
                d="M5,9h5.5A2.5,2.5,0,0,1,13,11.5v0A2.5,2.5,0,0,1,10.5,14H5a0,0,0,0,1,0,0V9A0,0,0,0,1,5,9Z"
              />
               
            </svg>
          </button>
          <button
            class="ql-italic"
            type="button"
          >
            <svg
              viewBox="0 0 18 18"
            >
               
              <line
                class="ql-stroke"
                x1="7"
                x2="13"
                y1="4"
                y2="4"
              />
               
              <line
                class="ql-stroke"
                x1="5"
                x2="11"
                y1="14"
                y2="14"
              />
               
              <line
                class="ql-stroke"
                x1="8"
                x2="10"
                y1="14"
                y2="4"
              />
               
            </svg>
          </button>
          <button
            class="ql-underline"
            type="button"
          >
            <svg
              viewBox="0 0 18 18"
            >
               
              <path
                class="ql-stroke"
                d="M5,3V9a4.012,4.012,0,0,0,4,4H9a4.012,4.012,0,0,0,4-4V3"
              />
               
              <rect
                class="ql-fill"
                height="1"
                rx="0.5"
                ry="0.5"
                width="12"
                x="3"
                y="15"
              />
               
            </svg>
          </button>
          <button
            class="ql-link"
            type="button"
          >
            <svg
              viewBox="0 0 18 18"
            >
               
              <line
                class="ql-stroke"
                x1="7"
                x2="11"
                y1="7"
                y2="11"
              />
               
              <path
                class="ql-even ql-stroke"
                d="M8.9,4.577a3.476,3.476,0,0,1,.36,4.679A3.476,3.476,0,0,1,4.577,8.9C3.185,7.5,2.035,6.4,4.217,4.217S7.5,3.185,8.9,4.577Z"
              />
               
              <path
                class="ql-even ql-stroke"
                d="M13.423,9.1a3.476,3.476,0,0,0-4.679-.36,3.476,3.476,0,0,0,.36,4.679c1.392,1.392,2.5,2.542,4.679.36S14.815,10.5,13.423,9.1Z"
              />
               
            </svg>
          </button>
        </span>
        <span
          class="ql-formats"
        >
          <button
            class="ql-list"
            type="button"
            value="ordered"
          >
            <svg
              viewBox="0 0 18 18"
            >
               
              <line
                class="ql-stroke"
                x1="7"
                x2="15"
                y1="4"
                y2="4"
              />
               
              <line
                class="ql-stroke"
                x1="7"
                x2="15"
                y1="9"
                y2="9"
              />
               
              <line
                class="ql-stroke"
                x1="7"
                x2="15"
                y1="14"
                y2="14"
              />
               
              <line
                class="ql-stroke ql-thin"
                x1="2.5"
                x2="4.5"
                y1="5.5"
                y2="5.5"
              />
               
              <path
                class="ql-fill"
                d="M3.5,6A0.5,0.5,0,0,1,3,5.5V3.085l-0.276.138A0.5,0.5,0,0,1,2.053,3c-0.124-.247-0.023-0.324.224-0.447l1-.5A0.5,0.5,0,0,1,4,2.5v3A0.5,0.5,0,0,1,3.5,6Z"
              />
               
              <path
                class="ql-stroke ql-thin"
                d="M4.5,10.5h-2c0-.234,1.85-1.076,1.85-2.234A0.959,0.959,0,0,0,2.5,8.156"
              />
               
              <path
                class="ql-stroke ql-thin"
                d="M2.5,14.846a0.959,0.959,0,0,0,1.85-.109A0.7,0.7,0,0,0,3.75,14a0.688,0.688,0,0,0,.6-0.736,0.959,0.959,0,0,0-1.85-.109"
              />
               
            </svg>
          </button>
          <button
            class="ql-list"
            type="button"
            value="bullet"
          >
            <svg
              viewBox="0 0 18 18"
            >
               
              <line
                class="ql-stroke"
                x1="6"
                x2="15"
                y1="4"
                y2="4"
              />
               
              <line
                class="ql-stroke"
                x1="6"
                x2="15"
                y1="9"
                y2="9"
              />
               
              <line
                class="ql-stroke"
                x1="6"
                x2="15"
                y1="14"
                y2="14"
              />
               
              <line
                class="ql-stroke"
                x1="3"
                x2="3"
                y1="4"
                y2="4"
              />
               
              <line
                class="ql-stroke"
                x1="3"
                x2="3"
                y1="9"
                y2="9"
              />
               
              <line
                class="ql-stroke"
                x1="3"
                x2="3"
                y1="14"
                y2="14"
              />
               
            </svg>
          </button>
        </span>
        <span
          class="ql-formats"
        >
          <button
            class="ql-clean"
            type="button"
          >
            <svg
              class=""
              viewBox="0 0 18 18"
            >
               
              <line
                class="ql-stroke"
                x1="5"
                x2="13"
                y1="3"
                y2="3"
              />
               
              <line
                class="ql-stroke"
                x1="6"
                x2="9.35"
                y1="12"
                y2="3"
              />
               
              <line
                class="ql-stroke"
                x1="11"
                x2="15"
                y1="11"
                y2="15"
              />
               
              <line
                class="ql-stroke"
                x1="15"
                x2="11"
                y1="11"
                y2="15"
              />
               
              <rect
                class="ql-fill"
                height="1"
                rx="0.5"
                ry="0.5"
                width="7"
                x="2"
                y="14"
              />
               
            </svg>
          </button>
        </span>
      </div>
      <div
        class="ql-container ql-snow"
      >
        <div
          class="ql-editor ql-blank"
          contenteditable="true"
          data-gramm="false"
        >
          <p>
            <br />
          </p>
        </div>
        <div
          class="ql-tooltip ql-hidden"
        >
          <a
            class="ql-preview"
            href="about:blank"
            rel="noopener noreferrer"
            target="_blank"
          />
          <input
            data-formula="e=mc^2"
            data-link="https://quilljs.com"
            data-video="Embed URL"
            type="text"
          />
          <a
            class="ql-action"
          />
          <a
            class="ql-remove"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
